src/walletConnect/saga.ts

```diff
@@ -331,12 +347,58 @@ function getSupportedChains() {
   })
 }
 
+export function* normalizeTransaction(rawTx: any, network: Network) {
+  const tx: TransactionRequest = { ...rawTx }
+
+  // Handle `gasLimit` as a misnomer for `gas`, it usually comes through in hex format
+  if ('gasLimit' in tx && tx.gas === undefined) {
+    tx.gas = isHex(tx.gasLimit) ? hexToBigInt(tx.gasLimit as Hex) : (tx.gasLimit as bigint)
+    delete tx.gasLimit
+  }
+
+  // re-calculate the feeCurrency and gas since it is not possible to tell from
+  // the request payload if the feeCurrency is set to undefined (native
+  // currency) explicitly or due to lack of feeCurrency support
+  if (network === Network.Celo) {
+    delete tx.gas
+    if ('feeCurrency' in tx) {
```

Comment:
apparently we can omit this check, because delete would't be upset if the property is not present

Reply:
yes....but then typescript complains 🙈 so i've left it like this